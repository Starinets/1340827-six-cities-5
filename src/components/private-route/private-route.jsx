import React from 'react';
import * as Type from '../../types';
import {Route, Redirect} from 'react-router-dom';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../store/selectors';
import {AuthorizationStatus} from '../../constants';

const PrivateRoute = (props) => {
  const {path, exact, render, authorizationStatus} = props;
  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(routeProps)
            : <Redirect to={`/login`} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: Type.STRING.isRequired,
  exact: Type.BOOLEAN.isRequired,
  path: Type.STRING.isRequired,
  render: Type.FUNCTION.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});

export default connect(mapStateToProps)(PrivateRoute);
