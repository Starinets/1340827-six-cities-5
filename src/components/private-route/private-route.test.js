import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {PrivateRoute} from './private-route';
import {
  emptyFunction,
  AuthorizationStatus
} from '../../test-data/test-data';

describe(`Component rendered correctly -> PrivateRoute`, () => {
  it(`with AuthorizationStatus.AUTH`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
        <PrivateRoute
          authorizationStatus={ AuthorizationStatus.AUTH }
          exact={ true }
          path={`/`}
          render={ emptyFunction }
        />
    );
    expect(tree).toMatchSnapshot();
  });

  it(`with AuthorizationStatus.NO_AUTH`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
        <PrivateRoute
          authorizationStatus={ AuthorizationStatus.NO_AUTH }
          exact={ true }
          path={`/`}
          render={ emptyFunction }
        />
    );
    expect(tree).toMatchSnapshot();
  });
});
