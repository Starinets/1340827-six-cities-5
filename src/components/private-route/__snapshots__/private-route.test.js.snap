// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendered correctly -> PrivateRoute with AuthorizationStatus.AUTH 1`] = `
<Route
  exact={true}
  path="/"
  render={[Function]}
/>
`;

exports[`Component rendered correctly -> PrivateRoute with AuthorizationStatus.NO_AUTH 1`] = `
<Route
  exact={true}
  path="/"
  render={[Function]}
/>
`;
